### PR TITLE
Addresses BACKLOG-40051: hide edit, delete, pause scheduler

### PR DESCRIPTION
**Pull request by mbrasil**
@pentaho/millenniumfalcon please review

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 9m
Build command:
```console
mvn clean verify -B -e -Daudit -amd -Djs.no.sandbox -pl core,ui
```
---
:ok_hand: All tests passed!
 
Tests run: **356**, Failures: **0**, Skipped: **12** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pentaho-scheduler-plugin/job/PR-150/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by Pancho7 — APPROVED** [review]

**Review by JoaoEnesGoncalves — APPROVED** [review]

File: core/src/main/java/org/pentaho/platform/web/http/api/resources/services/SchedulerService.java [hunk 1]
```diff
@@ -257,17 +257,12 @@ public Job updateJob( JobScheduleRequest scheduleRequest )
   public Job triggerNow( String jobId ) throws SchedulerException {
     Job job = (Job) getJob( jobId );
 
-    if ( getPolicy().isAllowed( SchedulerAction.NAME ) ) {
+    if ( isScheduleAllowed() || getSession().getName().equals( job.getUserName() ) ) {
```
Comment (afrjorge) [inline core/src/main/java/org/pentaho/platform/web/http/api/resources/services/SchedulerService.java:260]: this is to be tackled in **BACKLOG-40049**

File: core/src/main/java/org/pentaho/platform/web/http/api/resources/services/SchedulerService.java [hunk 2]
```diff
@@ -393,33 +388,31 @@ public String shutdown() throws SchedulerException {
   public JobState pauseJob( String jobId ) throws SchedulerException {
     Job job = (Job) getJob( jobId );
 
-    if ( isScheduleAllowed() || PentahoSessionHolder.getSession().getName().equals( job.getUserName() ) ) {
+    if ( isScheduleAllowed() || getSession().getName().equals( job.getUserName() ) ) {
```
Comment (afrjorge) [inline core/src/main/java/org/pentaho/platform/web/http/api/resources/services/SchedulerService.java:391]: this is to be tackled in **BACKLOG-40050**

**Review by afrjorge — APPROVED** [review]

**Pull request merged**